src/providers/settings.ts

```diff
@@ -1,70 +1,102 @@
 import { Injectable } from '@angular/core';
-
 import { Storage } from '@ionic/storage';
+import { Observable } from 'rxjs';
+import { Constants } from '../app/constants';
 
 /**
  * A simple settings/config class for storing key/value pairs with persistence.
  */
 @Injectable()
 export class Settings {
-  private SETTINGS_KEY: string = '_settings';
-
   settings: any;
 
-  _defaults: any;
-  _readyPromise: Promise<any>;
-
-  constructor(public storage: Storage, defaults: any) {
-    this._defaults = defaults;
-  }
-
-  load() {
-    return this.storage.get(this.SETTINGS_KEY).then((value) => {
-      if(value) {
+  settingsObservable: Observable<any> = Observable.fromPromise(this.storage.get(Constants.STORAGE_KEY_SETTINGS)
+    .then((value) => {
+      if (value) {
         this.settings = value;
-        this._mergeDefaults(this._defaults);
+        return this._mergeDefaults(this._defaults);
       } else {
         return this.setAll(this._defaults).then((val) => {
           this.settings = val;
         })
       }
-    });
+    })
+  ).share();
+
+
+  _defaults: any;
+
+  constructor(public storage: Storage, defaults: any) {
+    this._defaults = defaults;
   }
 
-  _mergeDefaults(defaults: any) {
-    for(let k in defaults) {
-      if(!(k in this.settings)) {
-        this.settings[k] = defaults[k];
+  _mergeDefaults(defaults: any): Promise<any> {
+    return new Promise((resolve, reject) => {
+      for (let k in defaults) {
+        if (!(k in this.settings)) {
+          this.settings[k] = defaults[k];
+        }
       }
-    }
-    return this.setAll(this.settings);
+      this.setAll(this.settings).then(result => {
```

Comment:
Curious why this was changed? `new Promise` used in this way is an anti pattern that loses the promise chain. the two methods should be identical except for the way the catch/reject is being handled...